mmv1/third_party/terraform/services/redis/resource_redis_cluster_test.go.tmpl

```diff
@@ -270,6 +270,70 @@ func TestAccRedisCluster_createUpdateDeletionProtection(t *testing.T) {
 	})
 }
 
+func TestAccRedisCluster_automatedBackupConfig(t *testing.T) {
+	t.Parallel()
+
+	context_enabled := map[string]interface{}{
+		"random_suffix":         acctest.RandString(t, 10),
+		"automated_backup_mode": "ENABLED",
+	}
+
+	context_diabled := map[string]interface{}{
+		"random_suffix":         context_enabled["random_suffix"],
+		"automated_backup_mode": "DISABLED",
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckRedisClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRedisCluster_automatedBackupConfig(context_enabled),
+			},
+			{
+				ResourceName:      "google_redis_cluster.cluster_abc",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccRedisCluster_automatedBackupConfig(context_diabled),
+			},
+			{
+				ResourceName:      "google_redis_cluster.cluster_abc",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+
+func testAccRedisCluster_automatedBackupConfig(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_redis_cluster" "cluster_abc" {
+  name                           = "tf-test-redis-abc%{random_suffix}"
+  shard_count                    = 1
+  region                         = "us-central1"
+  deletion_protection_enabled    = false
+  automated_backup_config {
+   automated_backup_mode         = "ENABLED"
```

Comment:
This should draw the value from the context rather than hardcode it, right?

Reply:
Normally yes, deletion_protection_enabled is set by default to true by the provider, so I need explicitly set it to false so it can be destroyed by the tests